src/main.ts

```diff
@@ -3,67 +3,83 @@ import { LightSheetOptions } from "./main.types.ts";
 import Sheet from "./core/structure/sheet.ts";
 import { CellInfo } from "./core/structure/sheet.types.ts";
 import Events from "./core/event/events.ts";
-
+import LightSheetHelper from "../utils/helpers.ts";
+import { DefaultRowCount, DefaultColCount } from "../utils/constants.ts";
 export default class LightSheet {
-  ui: UI;
+  #ui: UI;
   options: LightSheetOptions;
   sheet: Sheet;
   events: Events;
+  onCellChange?;
+  isReady: boolean = false;
 
   constructor(targetElement: Element, options: LightSheetOptions) {
     this.options = options;
+    this.options.defaultColCount = options.defaultColCount ?? DefaultColCount;
+    this.options.defaultRowCount = options.defaultRowCount ?? DefaultRowCount;
     this.events = new Events();
     this.sheet = new Sheet(this.events);
-    this.ui = new UI(
+    this.#ui = new UI(
       targetElement,
       this,
-      this.options.data.length,
-      this.options.data[0].length,
+      this.options.data?.length ?? this.options.defaultColCount,
```

Comment:
Passing `defaultColCount` here, but the parameter is `rowCount`?

Reply:
Yes its a mistake. thanks for pointing out

Reply:
> Shouldn't we pass both row and column count to the UI? It seems weird to require just one of them instead of neither or both.

@rafinutshaw can you comment on this? I'll approve regardless, just making sure there's a reason for it

Reply:
I just realized i dont even need row and col count in the UI class. This was just old code. so removed them form the constructor. I only need them to initialize the table from main.